server/src/instant/core.clj

```diff
@@ -105,6 +106,20 @@
   (stop)
   (start))
 
+(defn add-shutdown-hook []
+  (.addShutdownHook (Runtime/getRuntime)
+                    (Thread. (fn []
+                               (tracer/record-info! {:name "shut-down"})
+                               (tracer/with-span! {:name "stop-server"}
+                                 (stop))
+                               (tracer/with-span! {:name "stop-invalidator"}
+                                 ;; Hack to get the invalidator to shut down in
+                                 ;; dev. Otherwise the stream takes forever to close.
+                                 (when (= :dev (config/get-env))
```

Comment:
Question: should we do this in prod too? 

My thinking is there could be some situation where Instant goes down, and all database writes stop

Reply:
(discussed in person, but will recap here for posterity)

In prod, it should be fine if the close gets stuck because elasticbeanstalk will force it to shut down eventually after it sends the initial shutdown signal (assuming it sends one we can respond to and doesn't just send SIGKILL). Before that happens, eb will drain the connections and we call server/stop to force anyone who is connected to reconnect to a new server.